### DISCUSSION TITLE
Allow adding/editing of metadata when multiple nodes are selected

### DISCUSSION THREAD

**Discussion by DCON-b** [godotengine/godot-proposals#10995] [Editor]
Metadata is really powerful, but when you select multiple nodes in the editor, the ability to add metadata disappears, meaning you can only add metadata to a single node at a time. This makes adding metadata to multiple nodes rather tedious.

The editor should support adding/editing metadata to nodes when multiple nodes are selected, similar to how editing any other property in the editor will edit the relevant property for each selected node.

**Comment by Calinou** [comment]
One difficulty is that multiple nodes may have different metadata keys available (and/or different values to them, but it's less of a concern here). Should we only display a metadata key when **all** selected nodes have the metadata key in question set, or should we display it as soon as one of the nodes have the metadata key set? In this case, what should changing the metadata do? Should it automatically create the metadata on the other nodes where it doesn't exist yet, or only set it if already existing?

**Reply by DCON-b** [reply]
Personally, the only feature I want is to be able to add metadata to multiple nodes at once- meaning that if I select multiple nodes and add metadata, it will be added to each node. (Since most of my use cases, I am simply checking for the existence of the key, and the value is mostly irrelevant. In cases where the value **does** matter, I usually set the values by code, which doesn't require metadata to already exist on the node, and as such, nothing is needed from the editor).

However, it would be nice to have the ability to edit multiple nodes metadata at once, and it makes sense from a UX perspective to make this work with editing metadata as well. I would expect that only keys which are unanimously shared by the selected nodes would show in the editor, similar to how non-metadata properties are displayed when selecting multiple nodes in the editor as of today.

To answer your questions directly:

> Should we only display a metadata key when all selected nodes have the metadata key in question set

Yes.

> or should we display it as soon as one of the nodes have the metadata key set? In this case, what should changing the metadata do?

In this case, I would expect:

>  it automatically creates the metadata on the other nodes where it doesn't exist yet

**Comment by cpazzra** [comment]
To expand on the need for something like this with a real world use case, since I'm just now running into this myself.

I use [Cyclops Level Builder](https://github.com/blackears/cyclopsLevelBuilder) and more recently discovered a handy addon [Material Footsteps](https://github.com/COOKIE-POLICE/godot-material-footsteps) which uses a metadata to determine what sounds to play.
This is a handy system and would be event more convenient if I could add relevant metadata to selected nodes, rather than having to select then one at a time.

![image](https://github.com/user-attachments/assets/42360cab-6e28-4d94-8c04-1e648cd51b0b)

![image](https://github.com/user-attachments/assets/667122b9-56dd-47ab-a8d9-2d3fe98201b2)

I agree, when multi-selecting nodes, only shared metadata tags and the option to add a new metadata should be present. Maybe a tooltip or something to note this? But that seems the most straight forward to me.

**Comment by KnIfER** [comment]
Please try user-script addons for godot editor !

for example use my plugin https://github.com/KnIfER/Godot-Editor-Console

just run: 

```gdscript
for n in sns(): # sns stands for selected nodes
    n.set_meta("dir", "back")
```



Many minor issues can be solved using such editor plugin with dynamic scripts.   

It's very simple!   

If you open the plugin's snippets.gd, there's also full syntax highlighting and auto-completion.

**Comment by KnIfER** [comment]
found this https://github.com/ballerburg9005/godot-metadata-inspector  

but it's for godot 3
